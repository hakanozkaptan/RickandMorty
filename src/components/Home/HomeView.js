import React, { Component } from 'react';
import { connect } from 'react-redux';
import agent from '../../agent';
import { GET_DATA } from '../../constants/actionTypes';
import ListView from '../Shared/ListView';

const mapStateToProps = state => ({
  characters: state.characters.characters,
  page: state.characters.page,
  pageCount: state.characters.pageCount
});

const mapDispatchToProps = dispatch => ({
  fetchData: (payload) => dispatch({ type: GET_DATA, payload })
});

class HomeView extends Component {
  componentWillMount() {
    this.getData();
  }

  getData = () => {
    const currentPage = this.props.page;
    const getCharactersPromise = agent.Characters.all(currentPage);

    this.props.fetchData(getCharactersPromise);
  }

  loadMore = () => {
    this.getData();
  }

  render () {
    const {
      characters,
      loading,
      pageCount,
      page
    } = this.props;

    return (
      <div className="home-page">
        <div className="container page">
          <div className="row">
            <ListView
              characters={characters}
              loading={loading}
              pageCount={pageCount}
              currentPage={page}
              loadMore={this.loadMore}
            />
          </div>
        </div>
      </div>
    );
  } 
}


export default connect(mapStateToProps, mapDispatchToProps)(HomeView);
