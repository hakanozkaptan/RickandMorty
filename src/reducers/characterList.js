import { GET_DATA } from '../constants/actionTypes';

const defaultState = {
  page: 1,
  characters: [],
  characterCount: 0,
  pageCount: 0
};

export default (state = defaultState, action) => {
  switch (action.type) {
    case GET_DATA:
      return {
        ...state,
        characters: [...state.characters, ...action.payload.results],
        characterCount: action.payload.results.length,
        pageCount: action.payload.info.pages,
        page: state.page + 1
      };
    default:
      return state;
  }
};
