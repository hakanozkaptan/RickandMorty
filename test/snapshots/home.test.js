jest.mock('react-redux', () => require('react-redux-mock'));

import React from 'react';
import { MemoryRouter } from 'react-router';
import { __setState } from 'react-redux';
import HomeView from "../../src/components/Home/HomeView";
import Enzyme, { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import characterList from '../responses/characterList';

Enzyme.configure({ adapter: new Adapter() });

describe('home snapshot testing', () => {
    it('should render correctly if not loaded', () => {
        __setState({
            characters: {
                characters: []
            }
        });
        
        const componentRender = render(
            <MemoryRouter>
                <HomeView />
            </MemoryRouter>
        );

        expect(toJson(componentRender)).toMatchSnapshot();
    });

    it('should render correctly if loaded', () => {
        __setState({
            characters: {
                characters: characterList.results,
                page: 1,
                pageCount: 25
            }
        });

        const componentRender = render(
            <MemoryRouter>
                <HomeView />
            </MemoryRouter>
        );

        expect(toJson(componentRender)).toMatchSnapshot();
    });
});