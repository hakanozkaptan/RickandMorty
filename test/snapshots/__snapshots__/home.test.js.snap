// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home snapshot testing should render correctly if loaded 1`] = `
<div
  class="home-page"
>
  <div
    class="container page"
  >
    <div
      class="row"
    >
      <div
        class="container"
      >
        <ul
          class="cards"
        >
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/1"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Rick Sanchez
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/2"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Morty Smith
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/3"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Summer Smith
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/4"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Beth Smith
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/5"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Jerry Smith
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/6"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Abadango Cluster Princess
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/7"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Abradolf Lincler
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/8"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Adjudicator Rick
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/9"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Agency Director
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/10"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Alan Rails
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/11"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Albert Einstein
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/12"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Alexander
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/13"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Alien Googah
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/14"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Alien Morty
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/15"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Alien Rick
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/16"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Amish Cyborg
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/17"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Annie
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/18"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Antenna Morty
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/19"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Antenna Rick
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <a
                href="/character/20"
              >
                <div
                  class="LazyLoad"
                  style="height:300px"
                >
                  
                </div>
                <div
                  class="filler"
                >
                  
                </div>
                <div
                  class="card_content"
                >
                  <h2
                    class="card_heading"
                  >
                    Ants in my Eyes Johnson
                  </h2>
                </div>
              </a>
            </div>
          </li>
          <div
            class="loading"
          >
             
            <img
              alt="loader"
              src="loading.svg"
            >
              
            </img>
            Loading...
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`home snapshot testing should render correctly if not loaded 1`] = `
<div
  class="home-page"
>
  <div
    class="container page"
  >
    <div
      class="row"
    >
      <div
        class="container"
      >
        <ul
          class="cards"
        >
          
        </ul>
      </div>
    </div>
  </div>
</div>
`;
